Buildings/Fluid/Humidifiers/EvaporativeCoolers/Baseclasses/DirectCalculations.mo

```diff
@@ -0,0 +1,192 @@
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TWetBulIn(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Wet bulb temperature of the inlet air"
+    annotation (Placement(
+      visible=true,
+      transformation(
+        origin={-120,20},
+        extent={{-20,-20},{20,20}},
+        rotation=0),
+      iconTransformation(
+        origin={-120,60},
+        extent={{-20,-20},{20,20}},
+        rotation=0)));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput p(
+    final unit="Pa",
+    displayUnit="Pa",
+    final quantity="AbsolutePressure")
+    "Pressure"
+    annotation (Placement(
+      visible=true,
+      transformation(
+        origin={-120,-60},
+        extent={{-20,-20},{20,20}},
+        rotation=0),
+      iconTransformation(
+        origin={-120,-60},
+        extent={{-20,-20},{20,20}},
+        rotation=0)));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput dmWat_flow(
+    final unit="kg/s",
+    displayUnit="kg/s",
+    final quantity="MassFlowRate")
+    "Water vapor mass flow rate difference between inlet and outlet"
+    annotation (Placement(
+     visible=true,
+     transformation(
+       origin={120,0},
+       extent={{-20,-20},{20,20}},
+       rotation=0),
+     iconTransformation(
+       origin={120,0},
+       extent={{-20,-20},{20,20}},
+       rotation=0)));
+
+  Buildings.Fluid.Humidifiers.EvaporativeCoolers.Baseclasses.Xi_TDryBulTWetBul
+    XiOut(redeclare package Medium = Medium)
+    "Water vapor mass fraction at the outlet";
+
+  Buildings.Fluid.Humidifiers.EvaporativeCoolers.Baseclasses.Xi_TDryBulTWetBul
+    XiIn(redeclare package Medium =  Medium)
+    "Water vapor mass fraction at the inlet";
+
+  Modelica.Units.SI.Velocity vel
+    "Air velocity";
+
+  Modelica.Units.SI.ThermodynamicTemperature TDryBulOut(
+    displayUnit="degC")
+    "Dry bulb temperature of the outlet air";
+
+  Modelica.Units.SI.MassFlowRate mWat_flowOut
+    "Water mass flow rate at the outlet";
+
+protected
+  parameter Medium.ThermodynamicState sta_default=Medium.setState_pTX(
+    T=Medium.T_default, p=Medium.p_default, X=Medium.X_default)
+    "Default state of medium";
+  parameter Modelica.Units.SI.Density rho_default=Medium.density(sta_default)
+    "Density, used to compute fluid volume";
+
+equation
+  vel =abs(V_flow)/padAre;
+  eff = effCoe[1] + effCoe[2]*(dep) + effCoe[3]*(vel)  + effCoe[4]*(dep^2) +
+    effCoe[5]*(vel^2) + effCoe[6]*(dep*vel)  + effCoe[7]*(vel*dep^2) +
+    effCoe[8]*(dep*vel^3) + effCoe[9]*(dep^3*vel) + effCoe[10]*(vel^3*dep^2) +
+    effCoe[11]*(dep^3*vel^2);
+  TDryBulOut = TDryBulIn - eff*(TDryBulIn - TWetBulIn);
+  mWat_flowOut = XiOut.Xi[1]*V_flow*rho_default;
+  dmWat_flow = (XiOut.Xi[1] - XiIn.Xi[1])*V_flow*rho_default;
+  TDryBulIn = XiIn.TDryBul;
+  TWetBulIn = XiIn.TWetBul;
+  p = XiIn.p;
+  TDryBulOut = XiOut.TDryBul;
+  TWetBulIn = XiOut.TWetBul;
+  p = XiOut.p;
+  annotation (Documentation(info="<html>
```

Comment:
Change the section:
```
dmWat_flow = (XiOut.Xi[1] - XiIn.Xi[1])*V_flow*rho_default;
  TDryBulIn = XiIn.TDryBul;
  TWetBulIn = XiIn.TWetBul;
  p = XiIn.p;
  TDryBulOut = XiOut.TDryBul;
  TWetBulIn = XiOut.TWetBul;
  p = XiOut.p;
```
to be:
```
  connect(TDryBulIn, XiIn.TDryBul);
  connect(TWetBulIn, XiIn.TWetBul);
  connect(p, XiIn.p);
  connect(TDryBulOut, XiOut.TDryBul);
  connect(TWetBulIn, XiOut.TWetBul);
  connect(p, XiOut.p);
  dmWat_flow = (XiOut.Xi[1] - XiIn.Xi[1])*V_flow*rho_default;
```
and change the `TDryBulOut` to be a internal connector.

Reply:
Connections updated as per comment